gpu4pyscf/df/hessian/jk.py

```diff
@@ -0,0 +1,438 @@
+# Copyright 2021-2024 The PySCF Developers. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import ctypes
+import itertools
+import numpy as np
+from concurrent.futures import ThreadPoolExecutor
+import cupy
+from gpu4pyscf.df import int3c2e
+from gpu4pyscf.scf.int4c2e import libgint
+from gpu4pyscf.hessian.jk import _ao2mo
+from gpu4pyscf.lib import logger
+from gpu4pyscf.lib.cupy_helper import contract, cart2sph, reduce_to_device
+from gpu4pyscf.__config__ import _streams, _num_devices
+
+NROOT_ON_GPU = 7
+
+def _jk_task_with_mo1(dfobj, dms, mo_coeff, mo1s, occ_coeffs,
+                      with_j=True, with_k=True, hermi=0, device_id=0):
+    ''' Calculate J and K matrices with mo response
+        For CP-HF
+    '''
```

Comment:
`assert hermi == 1`
as the symmetry has been assumed in the input density matrix and the K construction

Reply:
Done